doc/fluid/design/dist_train/mpi_enabled_design.md

```diff
@@ -0,0 +1,33 @@
+#MPI-enabled PaddlePaddle Design doc
+
+# Background
+Now, PaddlePaddle Fluid with Distribution has relatively large network bottleneck, We want to use RDMA and GPUDriect to improve and solve it, so we enabled the features to PaddlePaddle with the help of MPI.
+
+We will introduce Open MPI API to PaddlePaddle, which can bring two benefits to PaddlePaddle:
+1. Enable RDMA with PaddlePaddle, which bring high-performance low latency networks.
+2. Enable GPUDriect with PaddlePaddle, which bring the highest throughput and lowest latency GPU read and write.
+
+## Execute args
+Launch the script using the ```mpirun``` launcher, For example: ```mpirun -np 3 -hosts node1,node2,node3 python train.py```. By doing this, We can number the actors (trainer/pserver/master) with o .. (n-1). The node's number is the Rank of the calling process in a group of comm (integer),  The MPI processes identify each other using a Rank ID. We have to create a mapping between PaddlePaddle's actors and their Rank ID so that we can communicate with the correct destinations when using MPI operations.
+    **We have to store the Rank ID and the mapping in global variables.**
+
+## New OP/MODULE
```

Comment:
Just list things need to be changed like:

- mpi_send_op
- mpi_serv_op
- modify transpiler to support using MPI or not
- compile arg to enable MPI support
- ...

then discuss the details.

Reply:
done